src/components/Form/Controls/EditInPlace/EditInPlace.tsx

```diff
@@ -205,59 +211,139 @@ export const EditInPlace = forwardRef<HTMLInputElement, Props>(
             </button>
           </div>
         </div>
-        {error && (
-          <div className={styles["caption-line"]}>
-            <ErrorIcon
-              className={classnames(
-                styles["caption-icon"],
-                styles["caption-icon-error"],
-              )}
-            />
-            <span
-              id={errorTextId}
-              className={classnames(
-                styles["caption-text"],
-                styles["caption-text-error"],
-              )}
-            >
-              {error}
-            </span>
-          </div>
-        )}
-        {saving && (
-          <div className={styles["caption-line"]}>
-            <InlineSpinner />
-            <span
-              className={classnames(
-                styles["caption-text"],
-                styles["caption-text-saving"],
-              )}
-            >
-              {savingLabel}
-            </span>
-          </div>
-        )}
-        {savedLabel && showSaved && (
-          <div className={styles["caption-line"]}>
-            <div
-              className={classnames(
-                styles["caption-icon"],
-                styles["caption-icon-saved"],
-              )}
-            >
-              <CheckIcon />
-            </div>
-            <span
-              className={classnames(
-                styles["caption-text"],
-                styles["caption-text-saved"],
-              )}
-            >
-              {savedLabel}
-            </span>
-          </div>
-        )}
+        <Labels
+          error={error}
+          errorTextId={errorTextId}
+          saving={saving}
+          savingLabel={savingLabel}
+          showSaved={showSaved}
+          savedLabel={savedLabel}
+          helpLabel={helpLabel}
+        />
       </form>
     );
   },
 );
+
+/**
+ * The labels that appear below the input box.
+ */
+interface LabelsProps {
+  /**
+   * The error message to display
+   */
+  error?: string;
+  /**
+   * The ID of the error text element
+   */
+  errorTextId: string;
+  /**
+   * True if the form is saving
+   */
+  saving: boolean;
+  /**
+   * The label to display while saving
+   */
+  savingLabel?: string;
+  /**
+   * The label to display when the form has been saved
+   */
+  savedLabel?: string;
+  /**
+   * True to show the saved label
+   */
+  showSaved?: boolean;
+  /**
+   * The label to display as a help message
+   */
+  helpLabel?: string;
+}
+
+function Labels({
+  error,
+  errorTextId,
+  saving,
+  savingLabel,
+  savedLabel,
+  showSaved,
+  helpLabel,
+}: LabelsProps): JSX.Element {
+  const labels: JSX.Element[] = [];
+  if (error) {
+    labels.push(
+      <div className={styles["caption-line"]} key={error}>
```

Comment:
This is probably okay, but I would go for static keys like `key="error"` because technically nothing is stopping two labels from having the same text. It also allows React to produce a more fine-grained diff when the text of a label changes.